### ISSUE TITLE
onItemClickListener issue

### ISSUE THREAD

**Issue by 8kt8** [mikepenz/FastAdapter#817]
## About this issue

Hello this is my override onItemClickListener, I did it simillar to your example(SimpleSubSexpandableItem), but more generic.

```
override var onItemClickListener: ((v: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int) -> Boolean)? =
        { view: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int ->

            adapter.fastAdapter.getItem(position)
            ifNotNull(view, expandableIconId) { v, id ->
                val rotationValue = if (item.isExpanded) 0f else 180f
                ViewCompat.animate(v.findViewById(id)).rotation(rotationValue).setDuration(200)
                    .start()
            }
            if (item.isExpanded) {
                expandAction(binding, adapter, item, position)
            } else {
                collapseAction(binding, adapter, item, position)
            }

            mOnClickListener?.invoke(view, adapter, item, position) ?: true
        }
        set(listener) {
            mOnClickListener = listener
        }
            
```
I am using fast adapter on Fragments not on Activities and everything working fine on first Fragment run, but when I go to the background or go to the next screen and back item from this listener return different item than adapter.fastAdapter.getItem(position). Item from listener have good view but isExpand property is always false.

When I am using adapter.fastAdapter?.getItem(position) to check isExpand property everythink workings fine. So I think there is something wrong with the item inserted in onItemClickListener.

```
override var onItemClickListener: ((v: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int) -> Boolean)? =
        { view: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int ->
 
            val adapterItem = adapter.fastAdapter?.getItem(position)
           override var onItemClickListener: ((v: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int) -> Boolean)? =
        { view: View?, adapter: IAdapter<ExpandableParentType>, item: ExpandableParentType, position: Int ->

            val adapterPosition = adapter.fastAdapter?.getItem(position)
            ifNotNull(view, expandableIconId, adapterPosition) { v, id, adapterPosition ->
                val rotationValue = if (adapterPosition.isExpanded) 0f else 180f
                ViewCompat.animate(v.findViewById(id)).rotation(rotationValue).setDuration(200)
                    .start()
            }
            if (adapterPosition.isExpanded) {
                expandAction(binding, adapter, item, position)
            } else {
                collapseAction(binding, adapter, item, position)
            }

            mOnClickListener?.invoke(view, adapter, item, position) ?: true
        }
        set(listener) {
            mOnClickListener = listener
        }
            
```

Version : 4.0.1
I discover this issue after your migration to Kotlin implementation.

**Comment by mikepenz** [comment]
@8kt8 so the item passed in from the listener is the one from the adapter. the same reference you should get. 

could it be that you check if it is expanded on the child element? or so?

Could you please provide a sample app to showcase this?

**Comment by bhautikkeraliya** [comment]
Respected Sir, 
i am also facing this issue.please help me.

**Comment by bhautikkeraliya** [comment]
> I fixed with use of "fastAdapter.withEventHook(new ...)"

**Comment by mikepenz** [comment]
Hi @8kt8  the message from @bhautikkeraliya is actually recommended. If you want a click listener on a view inside your item use a `EventHook` as it will ensure to only be created once and you will have no unnecessary listeners instantiated every time you scroll over the list.

The next version of the FastAdapter will come with a convenient extension functions to simplify this some more.

**Comment by camclendenin** [comment]
I had the same issue but I traced the cause of the click listener to the fact that the **mEnabled** property was set to false for some reason, but only in classes that I had converted to Kotlin. Additionally these classes use the @Parcelize if that is relevant. I can fix the issue by overriding **isEnabled** and always returning true, but this shouldn't be necessary. It's suspicious that it only happens after converting my models from Java to Kotlin.
